### PR TITLE
Migrate UNSAFE_componentWillReceiveProps

### PR DESCRIPTION
**Pull request by eloyekunle**
- [x] ./src/apps/expenses/components/CreateExpenseForm.js
- [x] ./src/apps/expenses/components/TransactionsWithData.js
- [x] ./src/components/CreateCollectiveForm.js
- [x] ./src/components/EditEventForm.js
- [x] ./src/components/EditCollectiveForm.js
- [x] ./src/components/InputField.js
- [x] ./src/components/InputTypeCreditCard.js
- [x] ./src/components/InputTypeLocation.js
- [x] ./src/components/MatchingFundWithData.js
- [x] ./src/components/PaymentMethodChooser.js
- [x] ./src/components/Subscriptions.js
- [x] ./src/components/UpdatesWithData.js
- [x] ./src/pages/banner-iframe.js

Fixes https://github.com/opencollective/opencollective/issues/1438

### CODE REVIEW

File: src/apps/expenses/components/CreateExpenseForm.js [hunk 1]
```diff
@@ -463,7 +463,7 @@ class CreateExpenseForm extends React.Component {
                   type="email"
                   name="paypalEmail"
                   key={`paypalEmail-${get(LoggedInUser, 'id')}`}
-                  defaultValue={this.state.expense.paypalEmail}
```
Comment (znarf) [inline src/apps/expenses/components/CreateExpenseForm.js]: Why this change?

File: src/components/PaymentMethodChooser.js [hunk 2]
```diff
@@ -67,29 +67,31 @@ class PaymentMethodChooser extends React.Component {
...
-    const { paymentMethodInUse, paymentMethodsList, editMode } = newProps;
+    const { paymentMethodInUse, paymentMethodsList, editMode } = this.props;
 
-    if (!paymentMethodInUse.name) {
-      // set state to modified
-      this.setState({ modified: true });
-      // If there was an existing card, select that
-      if (paymentMethodsList.length > 0) {
-        this.handleChange({ uuid: paymentMethodsList[0].uuid });
+    if (!prevProps.paymentMethodsList && paymentMethodsList) {
```
Comment (znarf) [inline src/components/PaymentMethodChooser.js]: Why introducing this test?

Comment (eloyekunle) [reply src/components/PaymentMethodChooser.js]: To avoid infinite loops, since we're setting state in `componentDidUpdate`. This ensures that the state will only be updated for this condition.

File: src/apps/expenses/components/CreateExpenseForm.js (hunk 1 above)
Comment (eloyekunle) [reply src/apps/expenses/components/CreateExpenseForm.js]: Initially, `paypalEmail` is `null` before the prop updates. `defaultValue` binds to the initial value during the first render so it doesn't change if the user is actually logged in and has a default email.
This change fixes that.
